src/compiler/sys.ts

```diff
@@ -1162,6 +1162,7 @@ namespace ts {
         useCaseSensitiveFileNames: boolean;
         write(s: string): void;
         writeOutputIsTTY?(): boolean;
+        getWidthOfTerminal(): number;

```

Comment:
When adding new method to api , it should be optional .. you would want to use some sort of default instead at usage site.. you shouldn’t need to change every host but only where you want to check formatting on host with different widths..

Reply:
good point.